pkg/services/extsvcauth/registry/service.go

```diff
@@ -32,17 +36,22 @@ func ProvideExtSvcRegistry(oauthServer oauthserver.OAuth2Server, saSvc *extsvcac
 // it generates client_id, secrets and any additional provider specificities (ex: rsa keys). It also ensures that the
 // associated service account has the correct permissions.
 func (r *Registry) SaveExternalService(ctx context.Context, cmd *extsvcauth.ExternalServiceRegistration) (*extsvcauth.ExternalService, error) {
+	// Record provider in case of removal
+	r.lock.Lock()
```

Comment:
I don't think this is mandatory. The plugin loading process should be single threaded. This is being future proof.

Reply:
I am getting a `panic` when running this:

```
panic: assignment to entry in nil map [recovered]
	panic: assignment to entry in nil map

goroutine 1 [running]:
github.com/grafana/grafana/pkg/cmd/grafana-server/commands.RunServer.func2()
	/github.com/grafana/grafana/pkg/cmd/grafana-server/commands/cli.go:96 +0x167
panic({0x4e6a640?, 0x652e380?})
	/usr/local/go/src/runtime/panic.go:914 +0x21f
github.com/grafana/grafana/pkg/services/extsvcauth/registry.(*Registry).SaveExternalService(0xc0024f6410, {0x658a008, 0x9975de0}, 0xc002c5abd0)
	/github.com/grafana/grafana/pkg/services/extsvcauth/registry/service.go:41 +0xaf
```